baseapp/baseapp.go

```diff
@@ -512,16 +499,11 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte, tx sdk.Tx) (result sdk
 		}
 		if !newCtx.IsZero() {
 			ctx = newCtx
-		}
-	}
+			// MIGHT NEED TO ADD result.Fee here
+			// like how the msg handler has: 		result.GasUsed = ctx.GasMeter().GasConsumed()
```

Comment:
I'm not sure if this is what you're referring to here, but there is a funny problem that is introduced by checkTx not running the transactions anymore. - We are don't deduct the fees in the checkTx state anymore - but we really should. If we don't, it means that one party who only has enough fees for one transaction can effectively spam by sending a bunch of transactions because none of them will fail during checkTx (although all of them past the first transaction SHOULD fail) - makes sense? Let's write a test for this situation